external/fv3fit/fv3fit/emulation/layers/architecture.py

```diff
@@ -20,52 +20,61 @@
 import tensorflow as tf
 from typing import Mapping, Optional, Sequence, Union, Any
 
+__all__ = ["ArchitectureConfig"]
 
-class CombineInputs(tf.keras.layers.Layer):
-    """Input tensor stacking with option to add a dimension for RNNs"""
 
-    def __init__(
-        self, *args, combine_axis: int = -1, expand_axis: Optional[int] = None, **kwargs
-    ):
-        """
+def combine_inputs(
+    inputs: Union[Sequence[tf.Tensor], Mapping[str, tf.Tensor]],
+    combine_axis: int = -1,
+    expand_axis: Optional[int] = None,
+) -> tf.Tensor:
+    """
         Args:
+            inputs: a datastructure of tensors to combine into a single tensor
             combine_axis: Axis to concatenate tensors along.  Note that if expand_axis
                 is specified, it is applied before concatenation.  E.g., combine_axis=1
                 and expand_axis=1 will concatenate along the newly created dimension.
             expand_axis: New axis to add to the input tensors
-        """
-        super().__init__(*args, **kwargs)
+    """
 
-        self._combine_axis = combine_axis
-        self._expand_axis = expand_axis
+    if isinstance(inputs, Mapping):
+        list_inputs = [inputs[key] for key in sorted(inputs)]
+    else:
+        list_inputs = list(inputs)
 
-    def _call_with_tensors(self, inputs: Sequence[tf.Tensor]) -> tf.Tensor:
+    if expand_axis is not None:
+        expanded_inputs = [
+            tf.expand_dims(tensor, axis=expand_axis) for tensor in list_inputs
+        ]
+    else:
+        expanded_inputs = list_inputs
 
-        if self._expand_axis is not None:
-            inputs = [
-                tf.expand_dims(tensor, axis=self._expand_axis) for tensor in inputs
-            ]
+    return tf.concat(expanded_inputs, axis=combine_axis)
 
-        return tf.concat(inputs, axis=self._combine_axis)
 
-    def call(
-        self, inputs: Union[Sequence[tf.Tensor], Mapping[str, tf.Tensor]]
-    ) -> tf.Tensor:
-        if isinstance(inputs, Mapping):
-            return self._call_with_tensors([inputs[key] for key in sorted(inputs)])
-        else:
-            return self._call_with_tensors(inputs)
+class RNNLayer(tf.keras.layers.Layer):
+    """Base class for RNN architectures
 
-    def get_config(self):
+    Scalar inputs (with singleton final dimensions) are passed to every element
```

Comment:
Could you put this under a `Note:`? I was confused for a while when I tried to read this as a description of the layer.